### PR TITLE
Make the creation of VulkanWindow platform agnostic by refactoring out Android specific surface creation logic.

### PR DESCRIPTION
**Pull request by chinmaygarde**
cc @abarth @jason-simmons

### CODE REVIEW

File: vulkan/vulkan_proc_table.h [hunk 1]
```diff
@@ -67,6 +66,10 @@ class VulkanProcTable {
   Proc<PFN_vkDestroyFence> destroyFence;
   Proc<PFN_vkWaitForFences> waitForFences;
 
+#if OS_ANDROID
```
Comment (abarth) [inline vulkan/vulkan_proc_table.h]: Do we need to include build_config.h to get this symbol?

Comment (chinmaygarde) [inline vulkan/vulkan_proc_table.h]: To access any of the function prototypes, we ned to import `vulkan_interface.h`. That has to include `build_config.h`.

**Comment by chinmaygarde** [comment]
Do GH approvals count as LGTM?

**Comment by abarth** [comment]
> Do GH approvals count as LGTM?

Yes, I think that makes sense.

**Pull request merged**